simpleclient_hibernate/src/main/java/io/prometheus/client/hibernate/HibernateStatisticsCollector.java

```diff
@@ -0,0 +1,249 @@
+package io.prometheus.client.hibernate;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.CounterMetricFamily;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.Statistics;
+
+/**
+ * Collect metrics from Hibernate statistics.
+ * <p>
+ * Usage example:
+ * <pre>
+ * new HibernateStatisticsCollector(sessionFactory, "default").register();
+ * </pre>
+ * If you are in a JPA environment, you can obtain the SessionFactory like this:
+ * <pre>
+ * SessionFactory sessionFactory =
+ *     entityManager.getEntityManagerFactory().unwrap(SessionFactory.class);
+ * </pre>
+ *
+ * @author Christian Kaltepoth
+ */
+public class HibernateStatisticsCollector extends Collector {
+
+  private static final List<String> LABEL_NAMES = Collections.singletonList("name");
+
+  private final Statistics statistics;
+  private final String name;
```

Comment:
Ah, OK. I wasn't aware that different collectors cannot produce the same metric name. But I just found the corresponding code in the registry.

Is there any specific reason why this isn't supported? I could imagine that if a single collector is responsible for monitoring "a thing" (like a DB connection pool), that it would make sense that a collector monitors just a single instance and that users can register the collector multiple times with different configuration. In this case the metric name would of cause be the same and the collector instances would just add different labels. Is there any reason why this isn't supported?

However, if you don't want to change the current behavior, I could implement something like:

```java
new HibernateStatisticsCollector()
    .add(sessionFactory1, "unit1")
    .add(sessionFactory2, "unit2")
    .register();